openfl/utilities/verifdataset/verifiable_dataset.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" This module contains the VerifiableDataset class. """
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.utilities.verifdataset.data_source import DataSource, DataSourceType
+from openfl.utilities.verifdataset.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
+
+
+def filter_non_serializable(obj):
+    """Filter out methods and non-serializable objects."""
+    serializable_dict = {}
+    for key, val in obj.__dict__.items():
+        if callable(val):
+            continue  # Skip methods
+        if isinstance(val, Enum):
+            val = val.value  # Convert Enum to its value
+        elif isinstance(val, Path):
+            val = str(val)  # Convert Path to string
+        serializable_dict[key] = val
+    return serializable_dict
+
+
+class VerifiableDataset:
+    """
+    This class represents a data set that can be verified.
+    It is containing multiple data sources and contains methods to compute the commitment,
+    validation, serialization and deserialization
+    """
+
+    def __init__(
+        self,
+        data_sources: List[DataSource],
+        base_dir: Path,
+        label: str,
+        metadata,
+        dataset_format: DatasetFormat,
+        valid_hash=None,
+    ):
+        self.data_sources = data_sources
+        self.label = label
+        self.metadata = metadata
+        self.base_dir = Path(base_dir)
+        self.dataset_format = dataset_format
+        self.hash = valid_hash
+
+    def _create_verbose_dataset_hash(self):
+        all_file_hashes = {
+            str(file_path.relative_to(self.base_dir)): ds.compute_object_hash(
+                self.base_dir / file_path
+            )
+            for ds in self.data_sources
+            for file_path in ds.enumerate_objects(self.base_dir)[0]
+        }
+        return all_file_hashes
+
+    def _create_concise_dataset_hash(self):
+        all_file_hashes = (
+            ds.compute_object_hash(file)
+            for ds in self.data_sources
+            for file in ds.enumerate_objects(self.base_dir)[0]
+        )
+        sorted_file_hashes = sorted(all_file_hashes)
+        joined_hashes = "".join(sorted_file_hashes)
+        return sha384(joined_hashes.encode()).hexdigest()
+
+    def create_dataset_hash(self):
+        """Create the hash of the dataset according to dataset format."""
+        if self.dataset_format == DatasetFormat.VERBOSE:
+            return self._create_verbose_dataset_hash()
+        if self.dataset_format == DatasetFormat.CONCISE:
+            return self._create_concise_dataset_hash()
+        raise ValueError("Unknown dataset format")
+
+    def _validate_verbose_dataset_info(self, valid_hash):
```

Comment:
Why do you need to pass `valid_hash` as a parameter, given that it's available as an object member under `self.hash`?